observability/logger.go

```diff
@@ -0,0 +1,74 @@
+package observability
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+
+	"github.com/TBD54566975/ftl/internal/log"
+)
+
+type logSink struct {
+	name   string
+	logger *log.Logger
+}
+
+var _ logr.LogSink = &logSink{}
+
+func (l *logSink) Init(info logr.RuntimeInfo) {
+}
+
+func (l logSink) Enabled(level int) bool {
+	return true
+}
+
+// otel uses the following mapping for level to our log.Level
+// 4 = Info
+// 8 = Debug
+// 1 = Warning
```

Comment:
This is a bit odd, but it's what I could see in their source. Maybe we just want to use `Tracef` for everything instead of following their levels?

https://github.com/open-telemetry/opentelemetry-go/blob/844b107e980ab1bd00ec2958ef735c4370a0c263/internal/global/internal_logging.go#L52-L70

Reply:
Error is probably 0 I assume? You'd have to dig deeper into the API to verify. Seems reasonable to map them?

Reply:
Found a reference in their test: https://github.com/open-telemetry/opentelemetry-go/blob/844b107e980ab1bd00ec2958ef735c4370a0c263/internal/global/internal_logging_test.go#L44-L51

Reply:
It also seems like errors go through the error handler. That's how I got the errors in the screenshot above.